utils_cv/detection/model.py

```diff
@@ -133,9 +95,83 @@ def get_pretrained_fasterrcnn(
     return model
 
 
+def _tune_mask_predictor(model: nn.Module, num_classes: int) -> nn.Module:
+    """ Tune mask predictor in the model. """
+    # get the number of input features of mask predictor from the pretrained
+    # model
+    in_features = model.roi_heads.mask_predictor.conv5_mask.in_channels
+    # replace the mask predictor with a new one
+    model.roi_heads.mask_predictor = MaskRCNNPredictor(
+        in_features,
+        256,
+        num_classes
+    )
+    return model
+
+
+def get_pretrained_fasterrcnn(
+    num_classes: int = None,
+    **kwargs,
+) -> nn.Module:
+    """ Gets a pretrained FasterRCNN model
+
+    Args:
+        num_classes: number of output classes of the model (including the background).
+
+    Returns
+        The model to fine-tine/inference with
+
+    For a list of all parameters see:
+        https://github.com/pytorch/vision/blob/master/torchvision/models/detection/faster_rcnn.py
+    """
+    # TODO - reconsider that num_classes includes background. This doesn't feel
+    #     intuitive.
+
+    # load a model pre-trained on COCO
+    model = _get_pretrained_rcnn(
+        fasterrcnn_resnet50_fpn,
+        **kwargs,
+    )
+
+    if num_classes:
+        model = _tune_box_predictor(model, num_classes)
+
+    return model
+
+
+def get_pretrained_maskrcnn(
+    num_classes: int = None,
+    **kwargs,
+) -> nn.Module:
+    """ Gets a pretrained Mask R-CNN model
+
+    Args:
+        num_classes: number of output classes of the model (including the
+            background)
+
+    Returns
+        The model to fine-tine/inference with
+
+    For a list of all parameters see:
+        https://github.com/pytorch/vision/blob/master/torchvision/models/detection/mask_rcnn.py
+
+    """
+    # load a model pre-trained on COCO
+    model = _get_pretrained_rcnn(
+        maskrcnn_resnet50_fpn,
+        **kwargs,
+    )
+
+    if num_classes:
+        model = _tune_box_predictor(model, num_classes)
```

Comment:
not understand?

Reply:
I think I was wrong. I assumed that "_tune_box_predictor()" does not need to get called since "_tune_mask_predictor()" was called. But guess both have to be called indeed. If so then ignore this comment.